src/views/authenticated-view/index.tsx

```diff
@@ -1,16 +1,25 @@
 import useAuthentication from 'hooks/use-authentication'
+import { AuthErrors } from 'types/auth'
+// import { signOut } from 'next-auth/react'
 import ErrorView from 'views/error-view-switcher'
 import { AuthenticatedViewProps } from './types'
+// import { signIn, SignInOptions } from 'next-auth/react'
 
 /**
  * A view that handles gating content behind authentication. Also handles
  * redirecting the user to the specified `redirectTo` page if they are not
  * authenticated.
  */
 const AuthenticatedView = ({ children }: AuthenticatedViewProps) => {
-	const { isAuthenticated, isAuthEnabled } = useAuthentication({
-		isRequired: true,
-	})
+	const { isAuthenticated, isAuthEnabled, session, signOut } =
+		useAuthentication({
+			isRequired: true,
+		})
+
+	if (session?.error === AuthErrors.RefreshAccessTokenError) {
+		signOut()
+		return null
+	}
```

Comment:
🤚🏻 Should this be handled in the Context since there are other views that use authentication for conditional rendering, but don't all use this view?

Reply:
Which context are you referring to?

Reply:
Oooh shucks! I mistakenly thought `useAuthentication` came from a Context. So scratch that specific idea!

Rather: can we move this to be handled inside of `useAuthentication` so it happens for all views that use the hook?

Reply:
I've moved the `signOut` call into `useAuthentication` and added a basic check to dedupe multiple requests to `signOut` (i.e. multiple rendered components on a page all concurrently attempting to `signOut`